src/main/java/Duke.java

```diff
@@ -1,10 +1,138 @@
+import java.util.Scanner;
+
 public class Duke {
+    public static final int MAX_NUMBER_OF_TASK = 100;
+
     public static void main(String[] args) {
+        printWelcomeMessage();
+        String command;
+        Task[] tasks = new Task[MAX_NUMBER_OF_TASK];
+        boolean isFinished = false;
+        Scanner input = new Scanner(System.in);
+        while (!isFinished) {
+            if (!input.hasNextLine()) {
+                break;
+            }
+            command = input.nextLine();
+            printSingleLine();
+            if (command.equals("bye")) {
+                printGoodbyeMessage();
+                isFinished = true;
+            } else if (command.equals("list")) {
+                printListOfTask(tasks);
+            } else {
+                updateListOfTask(command, tasks);
+            }
+        }
+    }
+
+    private static void printWelcomeMessage() {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        printSingleLine();
+        System.out.println("Hello! I'm Duke");
+        System.out.println("How can I help you?");
+        printSingleLine();
+    }
+
+    private static void printGoodbyeMessage() {
+        System.out.println("Bye. See you soon!");
+        printSingleLine();
+    }
+
+    private static void printListOfTask(Task[] tasks) {
+        if (Task.getNumberOfTask() == 0) {
+            System.out.println("Wow! I see that your list is empty");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < Task.getNumberOfTask(); i++) {
+                System.out.println((i + 1) + "." + tasks[(i + 1)].toString());
+            }
+        }
+        printSingleLine();
+    }
+
+    private static void printDetailsOfAddedTask(Task newTask) {
+        System.out.println("One more thing you got to do. Press on!");
+        System.out.println(newTask.toString());
+        System.out.println("Now you have " + Task.getNumberOfTask() + " tasks in the list.");
+        printSingleLine();
+    }
+
+    private static void updateTaskAsComplete(Task[] tasks, String description) {
+        int taskNumber = Integer.parseInt(description);
+        tasks[taskNumber].markAsDone();
+        printDetailsOfCompletedTask(tasks[taskNumber]);
+    }
+
+    private static void printDetailsOfCompletedTask(Task task) {
+        System.out.println("Nice. One more down!");
+        System.out.println(task.toString());
+        printSingleLine();
+    }
+
+    private static void updateListOfTask(String command, Task[] tasks) {
```

Comment:
The code is well written and although variables and method names are clear, maybe adding comments could make the code easier to understand.

Reply:
Ok, will take note on the point for adding comments. Thanks for the comments!